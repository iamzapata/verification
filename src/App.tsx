import { useEffect } from 'react'
import './App.css'
import { ChecksContainer } from './components/ChecksContainer'
import { useStore } from '@store'

function App() {
  const checks = useStore(state => state.checks)
  const { isLoading, hasError } = useStore(state => state.meta)
  const isValid = useStore(state => state.isValid)
  const getChecks = useStore(state => state.getChecks)
  const resultsSubmitted = useStore(state => state.resultsSubmitted)

  useEffect(() => {
    void getChecks()
  }, [])

  return (
    <div className="App">
      <ChecksContainer
        checks={checks}
        isLoading={isLoading}
        hasError={hasError}
        isValid={isValid()}
        resultsSubmitted={resultsSubmitted}
      />
    </div>
  )
}

export default App
