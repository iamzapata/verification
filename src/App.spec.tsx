import { render } from '@testing-library/react'
import App from './App'

describe('<App />', () => {
  it('should render', () => {
    expect(() => render(<App />)).not.toThrow()
  })

  it('should match snapshot', () => {
    const { container } = render(<App />)
    expect(container).toMatchSnapshot()
  })
})
