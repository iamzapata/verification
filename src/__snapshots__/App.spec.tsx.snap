// Vitest Snapshot v1

exports[`<App /> > should match snapshot 1`] = `
<div>
  <div
    class="App"
  >
    OK
  </div>
</div>
`;
